### ISSUE TITLE
Event identifier of an integer type

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-specification#4278]
**What are you trying to achieve?**

It may be good to allow having EventId of an integer type (instead of EventName which is always as string) for performance reasons.

**Additional context.**

In .NET the EventId is a integer with optional name.
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.eventid.-ctor?view=net-8.0

Also see OTel C++:
- https://github.com/open-telemetry/opentelemetry-cpp/blob/f69963f587b148d2eb20f7d30f8b4a1ccb184a6f/api/include/opentelemetry/logs/event_id.h

Related to:
- https://github.com/open-telemetry/opentelemetry-specification/issues/4260

**Comment by trask** [comment]
removed `tc-inbox` label since it's a `sig-issue`

**Comment by cijothomas** [comment]
Linking to prior attempts at having a durable identified for Logs(Events):
https://github.com/open-telemetry/semantic-conventions/issues/372
https://github.com/open-telemetry/semantic-conventions/pull/1339

**Comment by lmolkova** [comment]
Should we close it @pellared? EventName is now stable are we're past the point we could add an alternative. We could consider an attribute, but it's probably more practical to record number as string at this point.

**Comment by cijothomas** [comment]
Yes to closing. If we ever revisit this in future, it'll be via a *new* optional top-level field as we can't break EventName being string now.

> more practical to record number as string at this point.

SIGs who asked for this (C++,Rust, and .NET), would not want that, but would prefer integer for performance reasons (1 CPU instruction to compare when interfacing with kernel facilities like ETW/Tracepoints). Most likely, these SIG's would find alternate ways. For now, nothing is blocked and no active work to leverage Integer Ids!

(I'll come back when I get back to this area and hits blockers)
